### PR TITLE
Add TEST_TIME option for parallel testing

### PR DESCRIPTION
**Pull request by renfeiw**
- if TEST_TIME is not achievable, fail the build

Signed-off-by: renfeiw <renfeiw@ca.ibm.com>

### CODE REVIEW

File: buildenv/jenkins/JenkinsfileBase [hunk 1]
```diff
@@ -242,19 +249,23 @@ def setupParallelEnv() {
...
 	int num = params.NUM_MACHINES ? params.NUM_MACHINES.toInteger() : 1
-	def slaves = nodesByLabel(LABEL).size()
-	if (slaves == 1) {
-		echo "Only 1 slave machine available. Set num to 1"
-		return slaves
-	}
-	int limit = slaves / 2
+	int limit = getHalfSlaves()
 	if (num > limit) {
 		echo "Number of slave machines is ${slaves}. Number of machines cannot be greater than ${slaves} / 2 (=${limit}). Set num to ${limit}"
```
Comment (llxia) [inline buildenv/jenkins/JenkinsfileBase:254]: ${slaves} is moved to getHalfSlaves(). Please update the msg.

File: buildenv/jenkins/JenkinsfileBase [hunk 2]
```diff
@@ -242,19 +249,23 @@ def setupParallelEnv() {
...
 		num = limit
 	}
 	return num
 }
 
+def getHalfSlaves(){
+	def slaves = nodesByLabel(LABEL).size()
+	if (slaves == 1) {
+		echo "Only 1 slave machine available. Set num to 1"
```
Comment (llxia) [inline buildenv/jenkins/JenkinsfileBase:263]: Please remove the echo to avoid confusion.

**Comment by renfeiw** [comment]
Tested various combinations of TEST_TIME and NUM_MACHINES options in the internal builds:
| BUILD | TEST_TIME | NUM_MACHINES | RESULT |
--- | --- | --- | --- |
| Grinder_Sandbox/157  | "25" | "" | 5 lists |
| Grinder_Sandbox/159  | "2" | "" | failure as TEST_TIME is not possible |
| Grinder_Sandbox/160 | "" | "" | 1 list |
| Grinder_Sandbox/161 | "" | "3" | 3 lists |
| Grinder_Sandbox/162 | "2" | "3" | 3 lists |

File: buildenv/jenkins/JenkinsfileBase (hunk 1 above)
Comment (renfeiw) [reply buildenv/jenkins/JenkinsfileBase:254]: updated

File: buildenv/jenkins/JenkinsfileBase (hunk 2 above)
Comment (renfeiw) [reply buildenv/jenkins/JenkinsfileBase:263]: updated

File: buildenv/jenkins/JenkinsfileBase [hunk 3]
```diff
@@ -206,8 +215,6 @@ def setupParallelEnv() {
 						childParams << string(name: param.key, value: "${childTarget}")
 					} else if (param.key == "PARALLEL") {
 						childParams << string(name: param.key, value: "None")
-					} else if (param.key == "NUM_MACHINES") {
-						childParams << string(name: param.key, value: "1")
```
Comment (llxia) [inline buildenv/jenkins/JenkinsfileBase:210]: We need to set it back to empty (default) for child test builds

**Pull request merged**